### PR TITLE
Make C# Dev Kit Hot Reload discoverable while debugging

### PR DESCRIPTION
**Pull request by adamint**
## Description

C# Dev Kit already supplies Hot Reload for the normal `coreclr` sessions Aspire launches. Aspire does not need to modify the debug configuration or call into Dev Kit to enable it.

This change makes the disabled setting discoverable:

- logs the effective C# Dev Kit Hot Reload state for .NET project debug sessions
- shows one advisory per extension window when Dev Kit is installed and `csharp.experimental.debug.hotReload` is off
- opens VS Code settings filtered to that setting when the user selects **Open Settings**

It does not write third-party settings, persist notification state, call private Dev Kit commands, or change the generated debug configuration. The advisory is skipped for AppHost launches, `noDebug` launches, missing Dev Kit, and already-enabled Hot Reload.

## Validation

- Focused Hot Reload and .NET debugger coverage passes, including positive proof that an ordinary project debug launch invokes the advisory with the current diagnostics.
- The full extension suite passed after the reduction.
- Extension compile and lint completed successfully.
- A real Extension Development Host run confirmed that ordinary Aspire `coreclr` sessions receive C# Dev Kit Hot Reload services without Aspire-specific launch plumbing.

## Checklist

- [x] Feature is complete.
- [x] Tests cover the changed behavior.
- [x] No public API was added.
- [x] No new security guarantee is introduced.

### CODE REVIEW

**Comment by adamint** [comment]
## Review panel re-run — findings and fixes

I re-ran the full review panel because the earlier one was worthless: it ran against the *pipe-name-injection* design, and then I deleted that whole design after the A/B test falsified it. `hotReload.ts` was rewritten and never re-reviewed. That's the honest answer to "how was this not caught" — nothing looked at the code that's actually in the PR.

Full tier, 4 lanes, declared before dispatch (interactive UX + async lifecycle, ~1670 lines / 13 files):

| lane | model | effort | carried |
|---|---|---|---|
| architecture | `gpt-5.6-sol` | max | `vscode-extension` skill, hosting-core review patterns |
| edge cases | `claude-opus-5` | max | activation/trust/degradation matrix |
| domain | `gpt-5.6-sol` | max | `AGENTS.md`, `test-review-guidelines`, l10n rules |
| tester | `claude-opus-5` | high | proof-harness honesty |
| persistence *(escalation)* | `claude-opus-5` | xhigh | memento scope + write ordering |

Three lanes independently flagged that I'd left **persistence** out of the tier, so I added a fifth focused lane. It then corrected the other three: `.v1` isn't cargo cult (it matches `dashboardNotificationState.ts`), the reads are strict `=== true` so there's no corruption path, and the actual defect all three missed was **write ordering**. That's the one that mattered.

### Fixed (15)

The two real ones:

- **The exclusion was window-wide.** The prompt and the "Hot Reload is active" notice describe mutually exclusive states, so only one may speak — but showing the prompt suppressed the notice for *every later run in the window*, including the exact run the prompt just told the user to start. So you'd do what you were told and get silence. Now scoped to the launch.
- **The notice was recorded before it was shown.** Do Not Disturb, or a window closed mid-launch, burned the one time it's ever displayed.

The first fix had a bug of its own, caught by the incremental re-review: I scoped it on `runId`, which DCP generates *per `PUT /run_session`* — so sibling resources of one launch get different ones and the exclusion never fired between them. It's on `debugSessionId` now, which every resource of one app run shares. Same re-review found that a memento write failure would swallow the button the user pressed, and that my `globalState` pin didn't actually pin anything.

Rest: `showInformationMessage` sat outside the try/catch while a comment claimed the whole body was guarded; two floating thenables; a TDZ landmine; the prompt could offer a setting no installed extension declares (dead key, reported as success); the log claimed Hot Reload "covers" resources launched with `noDebug`.

And one test that was lying: "project launch succeeds when C# Dev Kit throws" threw from `isLimitedActivation`, which the code stopped reading three commits earlier. It passed with the defensive catch deleted. Replaced with two that throw from what's actually read — both die when the catch goes.

### Declined, with reasons

- **Delete `hotReloadNotificationState.ts`** (arch) — 17 lines matching `dashboardNotificationState.ts`. Consistency wins.
- **`getCommands()` check before offering "Show Hot Reload Output"** (arch) — the existing catch already degrades with a log line. Adding an await + a second failure mode inside fire-and-forget, to cover Dev Kit being installed *and* missing its own command, isn't worth it.
- **Gate the notice on confirmed session start** (arch) — needs a session-correlation layer for a cosmetic edge. The write-ordering half was the real bug and that's fixed.
- **`workspaceTrusted` is unreachable** (edge) — verified true: `package.json` declares no `capabilities`, so VS Code disables the extension outright in Restricted Mode. Keeping it anyway. It's the semantically correct predicate, costs one property read, and it's what keeps this correct if `capabilities.untrustedWorkspaces` ever gets declared.
- **`setKeysForSync`** (persistence) — never called anywhere in this extension. Pre-existing repo-wide gap, out of tier.

### Evidence

Every guard added here was mutation-tested — revert it, and *exactly* the test written for it fails, nothing else. That's 7 guards now: run scoping, write ordering, contributed-setting check, defensive catch (2 tests), launch scoping, selection-survives-storage-failure.

Re-proved end to end against real C# Dev Kit 3.29.195 after the changes, 3 .NET resources:

```
[proof] before edit: api  body='BEFORE-EDIT-1' pid=99474
[proof] after hot reload: api  body='AFTER-EDIT-1' pid=99474
[proof] after hot reload: api2 body='AFTER-EDIT-2' pid=99473
[proof] after hot reload: api3 body='AFTER-EDIT-3' pid=99472
ignored for EnC: 0
```

PIDs unchanged is the whole assertion — that's what separates an applied delta from a silent restart. And exactly one notice for three resources, which is the launch scoping working in the real product rather than in a stub.

1421 unit tests passing, tsc + eslint clean.

### Gaps I'm not pretending are closed

- The tester lane never returned (still running at 53 min when I stopped waiting), so its verdict on whether the proof harness could pass while hot reload is broken isn't in here. My own answer: the harness asserts body-changed **and** pid-unchanged **and** zero EnC-ignored, so a restart fails it and a no-op fails it. I think that's sound but it isn't an independent check.
- `extension.ts` wiring still isn't executed by any test. It's now a type error to hand `initializeHotReloadPromptState` anything but a context, which is stronger than the test I'd have written, but it isn't coverage.

**Comment by adamint** [comment]
Tester lane finally came back (55 min), so closing the gap I flagged above.

**Its verdict on harness honesty: yes.** The reasoning is better than mine — `/pid` is a *separate route the edit never touches*, read from the same base URL after the body flipped. A restart can't pass it (DCP hands out a new pid and usually a new port, so the fetch fails outright), a cache can't pass it (that pins the old body), and a race can't pass it (`waitFor` only accepts an exact match). The 20s settle is *before* the edit and exists to stop a harness-side Roslyn baseline race; the post-edit window is 180s of active polling, so it isn't masking anything.

It also caught something I'd missed: **the harness invoked `csdevkit.debug.hotReload`, so it proved the button, not the save gesture the notice text leads with.** That's the claim most users will actually exercise, and it was unproven. Fixed — re-ran with the button call disabled, edits applied by save alone, at the same HEAD as everything else:

```
[proof] SAVE-ONLY mode: NOT invoking csdevkit.debug.hotReload
[proof] before edit: api  body='BEFORE-EDIT-1' pid=5964
[proof] after hot reload: api  body='AFTER-EDIT-1' pid=5964
[proof] after hot reload: api2 body='AFTER-EDIT-2' pid=5966
[proof] after hot reload: api3 body='AFTER-EDIT-3' pid=5965
ignored for EnC: 0
```

Its other caveat is right and I'm not fixing it because it isn't a defect: the proof shows Aspire's `coreclr` sessions inherit Hot Reload, not that this PR *caused* that. Correct — the PR only detects and offers. That's the point of the whole simplification.

No-degradation E2E on real VS Code with the Aspire VSIX only (no C# extension, no Dev Kit): 5 passing, 2 failing. Both failures are `ElementNotInteractableError` in `InputBox.clear` inside `openAspireView` — Selenium command-palette flake, no product assertion reached. The load-bearing one passed: a .NET project resource reaches **Running** with neither extension installed (440s under load ~17).

File: extension/src/debugger/hotReload.ts [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+        settingEnabled: isHotReloadSettingEnabled(),
+        reloadOnSaveEnabled: isHotReloadOnSaveEnabled()
+    };
+}
+
+/**
+ * Whether Hot Reload is expected to apply to .NET project resources in this window.
+ */
+function isHotReloadExpected(diagnostics: HotReloadDiagnostics): boolean {
+    return diagnostics.devKitInstalled && diagnostics.workspaceTrusted && diagnostics.settingEnabled;
```
Comment (adamint) [inline extension/src/debugger/hotReload.ts]: I think this needs the same contribution check as the prompt. If Dev Kit renames the experimental key but a stale user value is still true, the notice says Hot Reload is enabled even though nothing reads that key. Could we gate `isHotReloadExpected` on `isHotReloadSettingContributed()` and make the missing-key diagnostic once-per-window so every resource does not repeat it?

File: extension/src/test/hotReload.test.ts [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+            return update;
+        }
+
+        test('does not offer to enable a setting no installed extension declares', async () => {
+            // The gate is explicitly experimental, and experimental settings get renamed. Offering it
+            // after a rename would write a key nothing reads into the user's settings and then report
+            // success - a promise Aspire cannot keep.
+            initializeHotReloadPromptState({ globalState: createTestMemento() });
+            const prompt = stubPrompt('Enable Hot Reload');
```
Comment (adamint) [inline extension/src/test/hotReload.test.ts]: These selections should use `enableHotReloadLabel` and `dontShowAgainLabel`, not English literals. Production compares against localized values, so under a localized Extension Host the stub returns a different string and these tests exercise the dismissal path instead of the intended branch.

File: extension/src/debugger/hotReload.ts [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+            // Recorded only once the notification has actually been through the user, so a notice
+            // that was never delivered — Do Not Disturb, a window closed mid-launch — does not burn
+            // the one time this is ever shown. `hotReloadNoticeShownThisWindow` above already stops
+            // concurrently launching resources from stacking notices while this is pending.
+            //
+            // Caught separately from the action below: this is bookkeeping, and a storage failure
+            // must not swallow the button the user actually pressed.
+            try {
+                await hotReloadPromptState?.update(hotReloadSessionNoticeShownKey, true);
```
Comment (adamint) [inline extension/src/debugger/hotReload.ts]: This records the once-ever notice before the requested action succeeds. The live Extension Host probe had Dev Kit installed but `csdevkit.debug.showHotReloadPanel` was not registered yet, so clicking the button did nothing and permanently suppressed the notice. Could we persist the flag only after the action succeeds, or surface the failure?

File: extension/src/debugger/hotReload.ts [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+    extensionLogOutputChannel.info(
+        `Hot Reload state for ${resourceName}: workspaceTrusted=${diagnostics.workspaceTrusted}, ` +
+        `csharp.experimental.debug.hotReload=${diagnostics.settingEnabled}, ` +
+        `csharp.debug.hotReloadOnSave=${diagnostics.reloadOnSaveEnabled}`);
+
+    if (!diagnostics.workspaceTrusted) {
+        extensionLogOutputChannel.info(
+            'The workspace is not trusted, so C# Dev Kit activates in limited mode and Hot Reload is unavailable.');
+    }
+
```
Comment (adamint) [inline extension/src/debugger/hotReload.ts]: When the workspace is untrusted, enabling the machine setting still cannot make Hot Reload work until trust is granted. Could we only emit this setting advice when `workspaceTrusted` is true so the diagnostics do not give the user an ineffective next step?

File: extension/src/debugger/languages/dotnet.ts [hunk 5]
```diff
@@ -554,6 +555,48 @@ export function createProjectDebuggerExtension(dotNetServiceProducer: (debugSess
...
+                // created, so awaiting it would stall the resource behind an advisory message.
+                //
+                // `launchOptions.debugSessionId` identifies the Aspire launch this resource belongs to, and every
+                // sibling resource of the same app run carries the same one. Both messages take it so
+                // that only one of them can speak for a launch, without suppressing the other for the
+                // rest of the window: after the user enables the setting here and restarts debugging
+                // as instructed, that next launch is when the notice should explain what Hot Reload
+                // now covers. Not `runId` — DCP generates that per `PUT /run_session`, so it differs
+                // between sibling resources of a single launch.
+                void promptToEnableHotReloadIfNeeded(hotReloadDiagnostics, isDebugSession, launchOptions.debugSessionId)
```
Comment (adamint) [inline extension/src/debugger/languages/dotnet.ts]: This launch key is not shared in production: `AspireDcpServer` passes the resource-specific `dcpId`, so sibling resources get different values and the test fabricates a contract the server does not provide. Could we pass the parsed parent Aspire debug-session id instead and add coverage with distinct resource ids under one session?

File: extension/src/debugger/hotReload.ts [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+ *
+ * The prompt and the notice describe mutually exclusive states, so at most one of them may speak for
+ * a given launch. Keyed on the Aspire debug session, which every resource of one app run shares —
+ * not on `runId`, which DCP generates per `PUT /run_session` and is therefore different for each
+ * sibling resource. Keying on `runId` would let the prompt speak for one project and the notice for
+ * the next, telling the user in the same breath to restart debugging and that Hot Reload is already
+ * on. Not window-wide either: after the user enables the setting and starts debugging again, that
+ * next launch is exactly when the notice should explain what Hot Reload now covers.
+ */
+let hotReloadMessageShownForLaunchId: string | undefined;
```
Comment (adamint) [inline extension/src/debugger/hotReload.ts]: A single remembered launch id still cannot model two Aspire sessions starting concurrently. If launch A is waiting on the enable prompt and launch B starts with the setting off, a later B resource can see the newly written global value and show the active notice even though earlier B resources started without Hot Reload. I think we need a per-launch snapshot/map of the initial state, or to drop the active notice.

File: extension/src/debugger/hotReload.ts (hunk 1 above)
Comment (adamint) [reply extension/src/debugger/hotReload.ts]: Fixed. `isHotReloadExpected` now gates on the contributed setting check, and the missing-setting diagnostic is only logged once per window.

File: extension/src/test/hotReload.test.ts (hunk 2 above)
Comment (adamint) [reply extension/src/test/hotReload.test.ts]: Fixed. These tests now select the localized labels instead of English literals.

File: extension/src/debugger/hotReload.ts (hunk 3 above)
Comment (adamint) [reply extension/src/debugger/hotReload.ts]: Fixed. The notice is persisted only after the output command succeeds; a missing command now logs the failure and leaves the notice available.

File: extension/src/debugger/hotReload.ts (hunk 4 above)
Comment (adamint) [reply extension/src/debugger/hotReload.ts]: Fixed. The setting advice is only emitted for trusted workspaces.

File: extension/src/debugger/languages/dotnet.ts (hunk 5 above)
Comment (adamint) [reply extension/src/debugger/languages/dotnet.ts]: Fixed. Hot Reload groups on the owning `AspireDebugSession.debugSessionId`, while the raw resource `dcpId` remains the configuration and notification-routing identity. The regressions use distinct resource ids under one parent session.

File: extension/src/debugger/hotReload.ts (hunk 6 above)
Comment (adamint) [reply extension/src/debugger/hotReload.ts]: Fixed. The initial Hot Reload state is snapshotted per parent Aspire launch id, so concurrent launches cannot observe each other’s later setting changes.

File: extension/src/test/hotReload.test.ts [hunk 7]
```diff
@@ -0,0 +1,851 @@
...
+        };
+
+        // The notice body is fire-and-forget so it cannot block the launch path, so let the
+        // microtask queue drain before asserting.
+        const flush = async () => { await new Promise(resolve => setTimeout(resolve, 5)); };
+
+        function stubContributedHotReloadSetting(): void {
+            sinon.stub(vscode.workspace, 'getConfiguration').returns({
+                inspect: () => ({ key: 'hotReload', defaultValue: false })
+            } as unknown as vscode.WorkspaceConfiguration);
```
[the remark this answers is not in the document]
Comment (adamint) [reply extension/src/test/hotReload.test.ts]: Fixed in 6cab7f7075. You were right, and the failure mode was worse than "some tests fail": the unit-test host installs no Dev Kit, so `inspect('hotReload')` reported no contributed default, `isHotReloadExpected()` returned early, and the silence assertions were passing for a reason that had nothing to do with the behavior they claim to pin.

The setting contribution is extension-host state, not user configuration, so it is now modelled explicitly rather than left to whatever happens to be installed. `createHotReloadTestConfiguration` in `test/common.ts` supplies that shape, and every test that expects an active notice stubs it — as does every test that expects silence, so those now prove silence with the gate open rather than with it accidentally closed.

The contribution flag is a parameter rather than always-on, because the uncontributed case is a real state worth testing: `{ contributed: false }` is what a renamed experimental key looks like.

That led somewhere I had not looked. With the gate stubbed consistently, the disabled-state log was telling the user to set `csharp.experimental.debug.hotReload` in user settings even when no installed extension contributes it, while `promptToEnableHotReloadIfNeeded` was refusing to write that same key for exactly that reason. Contradictory advice pointing at a fix that cannot work. Both now share `isHotReloadSettingContributedWithDiagnostic()`, and `does not tell the user to set a Hot Reload key that nothing contributes` covers it.

**Comment by adamint** [comment]
## Review findings addressed — 90c313c650

Three findings, one of which was raised twice and I had not acted on.

### The disabled-state log contradicted the enable prompt

Raised on `hotReload.ts:191` and again on `:192`. The gate is explicitly `experimental`, so a rename leaves it reading back as disabled — and in that state the log told the user to set `csharp.experimental.debug.hotReload` in user settings while `promptToEnableHotReloadIfNeeded` refused to write that same key, for exactly that reason. Contradictory advice pointing at a fix that cannot work.

Both now share `isHotReloadSettingContributedWithDiagnostic()`, so a renamed key produces only the accurate missing-contribution diagnostic. One line of shipping code.

### The AppHost guard was only pinned against notifications

The `isApphost` guard landed in 5066cae302, but `AppHost launch leaves the Hot Reload advisory for the first .NET resource` only asserted `notification.called === false`. Moving the guard below `logHotReloadDiagnostics` would have kept that assertion green while the AppHost reported itself as Hot Reload-covered. The test now asserts log silence too.

The suggestion that the polyglot regression exercise that path was the right one, and it closed a genuine gap. A polyglot Aspire app still has a .NET AppHost, and it launches through the same project debugger a .NET resource would — so it is the one path by which an app with **no** .NET resources could still surface Hot Reload UI. Two tests added, both driving the real project debugger:

```text
✔ the .NET AppHost of a polyglot app shows and logs nothing about Hot Reload
✔ a full polyglot app, AppHost included, stays silent
```

The second composes them in launch order, because the AppHost starts first and every guard here is once-per-window state an earlier launch could burn.

`TestDotNetService` moved from `dotnetDebugger.test.ts` into `test/common.ts` rather than being duplicated, so both suites drive the launch path the same way.

**Mutation tested** — each deliberate regression was introduced and confirmed to fail:

| Mutation | Result |
|---|---|
| Remove the AppHost guard entirely | 3 failing |
| Move the guard *below* the diagnostics logging | 3 failing |
| Drop the contribution check from the disabled-state log | 1 failing |

The second is the one that matters: it passed before this change.

### Concurrent DCP ids compared positionally

`aspireDcpServer.test.ts:69`. Both PUTs are in flight at once, so Express may service them in either order — the test was nondeterministic even when both ids were preserved correctly. Now compares sorted copies; the invariant is that each raw id survives distinctly, not the order they arrived in.

### Validation

**1516 passing, 1 failing.** The failure is `AppHost discovery > workspace project fallback checks projects beyond bounded file-search batch` — a 20s timeout in `appHostDiscovery.test.ts`, a file this PR does not touch. It passes in isolation (67/67, exit 0) and passed a full run on the previous commit, so it is load flake on my machine rather than a regression here. Test count went 1514 → 1517 with the 3 added tests.

Also verified: `tsc`, `eslint`, and `webpack` all clean.

**Comment by adamint** [comment]
## Review — 90c313c650

I went back through this end to end after pushing the review fixes, reading the shipping code rather than the description. **No blocking issues.** Recording what I actually verified, and the things a future reader should know rather than rediscover.

### What I checked, and how

**The prompt and the notice cannot both speak for one launch.** This is the claim most likely to be wrong, because `dotnet.ts` calls the prompt *un-awaited* and then calls the notice synchronously on the next line — which reads like a race.

It is not one, and the reason is worth stating because it is load-bearing and invisible: `promptToEnableHotReloadIfNeeded` sets `launchState.messageShown = true` **before its first `await`**, so the synchronous prefix has already run by the time `void` returns. The notice, called next, observes it. I walked the four reachable orderings:

| State at launch | Prompt | Notice |
|---|---|---|
| Setting off, not yet prompted this window | prompts, sets `messageShown` | returns on `messageShown` |
| Setting off, already prompted this window | returns early, `messageShown` untouched | returns on `isHotReloadExpected` (setting is off) |
| Setting on | returns early on `settingEnabled` | announces |
| User accepted in launch A, launch B starts | returns early on `settingEnabled` | announces — which is the intended "you were told to restart debugging; here is what it covers" |

The third and fourth are the ones that would produce a self-contradicting pair of notifications if the ordering were wrong.

**Nothing here can break a .NET debug session.** The entire block is inside `try`/`catch`, both notifications are fire-and-forget with attached `.catch`, and `launchOptions.debugSession` being undefined degrades to a warn rather than throwing on the launch path. `project launch succeeds when reading Hot Reload state throws` and `project launch is not blocked while the Hot Reload prompt is waiting for an answer` pin both halves.

**The per-launch snapshot is the right call.** The gate is machine-scoped and therefore global, so two Aspire apps starting concurrently can observe it change underneath them. `getHotReloadDiagnosticsForLaunch` freezes it on the first resource, which is what stops a launch from turning from "disabled" to "active" halfway through its own resources.

**The `Global` write target is correct and non-obvious.** `"scope": "machine"` means VS Code silently discards a workspace-scoped write — the prompt would report success while Hot Reload stayed off. `writes the enable at machine scope` pins it.

### Things worth knowing (not defects)

- **`isHotReloadSettingContributedWithDiagnostic()` reads live configuration**, while every other input to the decision comes from the per-launch snapshot. It is the one unsnapshotted value. Fine in practice — an extension cannot add or drop a contributed setting mid-launch without reloading the window — but it is an asymmetry, not an oversight to be "fixed" later without thinking.
- **`hotReloadLaunchStates` evicts the oldest entry at 64.** An in-flight launch could in principle be evicted and recreated, re-arming `messageShown`. That needs 64 concurrent launches in one window, so it is not worth code, but it is the failure mode if the bound is ever lowered.
- **Accepting the prompt also suppresses it permanently.** A user who later turns the setting back off is never re-offered. I think that is right — they know the feature exists now, and the alternative re-nags someone who made a deliberate choice.
- **`logHotReloadDiagnostics` writes a state line per .NET resource per launch** for every Dev Kit user. Deliberate, and the reason is in the comment: it is the only thing that answers "was *this* project covered?" for a user who dismissed the one-time notice.
- **The E2E shard adds two CI jobs.** Six of its seven tests are cheap structural assertions; only `runs a .NET project resource to Running…` starts an AppHost.

### On the shape of the change

The strongest thing about this PR is what it does *not* do. Shipping code is one new file plus about fifty lines across three existing ones. Nothing modifies a debug configuration, nothing calls a Dev Kit export, and no setting, command, or menu is contributed — the last of which is asserted, not assumed, because redeclaring another extension's machine-scoped setting produces a duplicate settings-UI entry and can change how the value resolves.

The E2E environment installs only the Aspire VSIX, which makes it simultaneously the configuration that must not regress and a real no-.NET-tooling polyglot environment. That is a better fixture than one built on purpose would have been.

### Validation

**CI: 338 checks complete, 0 failures** on `90c313c650`, including `Run VS Code extension unit tests (Windows)` — a clean host with no Dev Kit installed, which is precisely the environment the stubbing finding was about.

**Locally: 1516 passing.** The one failure is a 20s timeout in `appHostDiscovery.test.ts`, a file this PR does not touch; it passes in isolation (67/67) and passed a full run on the previous commit, so it is load flake on my machine. `tsc`, `eslint`, and `webpack` clean.

Still outstanding before this leaves draft: **screenshots of the two notifications**, and the `Fixes #` line in the description.

**Review by mitchdenny — APPROVED** [review]

File: extension/src/debugger/languages/dotnet.ts [hunk 8]
```diff
@@ -610,6 +611,21 @@ export function createProjectDebuggerExtension(dotNetServiceProducer: (debugSess
             if (launchOptions.isApphost && profileName) {
                 debugConfiguration.env['DOTNET_LAUNCH_PROFILE'] = profileName;
             }
+
+            if (!launchOptions.isApphost && debugConfiguration.noDebug !== true) {
+                // C# Dev Kit and vsdbg provide Hot Reload for the ordinary coreclr launch. Aspire only
+                // reports their effective settings and points users at the setting when it is disabled.
+                try {
+                    const hotReloadDiagnostics = getHotReloadDiagnostics();
+                    logHotReloadDiagnostics(`${projectPath} (run ${debugConfiguration.runId})`, hotReloadDiagnostics);
```
Comment (ellahathaway) [inline extension/src/debugger/languages/dotnet.ts:620]: The helper's formatting is covered in `hotReload.test.ts`, but I don't think the launch wiring is. The stub isn't retained, and the ordinary-launch test only checks the diagnostics and advisory calls, so removing `logHotReloadDiagnostics()` here would leave the focused suite green. Consider retaining the stub and asserting the resource/run identifier and diagnostics in the ordinary project launch test.

**Review by ellahathaway — APPROVED** [review]
I reviewed the C# Dev Kit Hot Reload diagnostics and advisory flow, the .NET debugger launch integration, localization, and regression coverage.

I didn't find any correctness or architectural issues. I left one nonblocking test-coverage comment about pinning the diagnostic logging call site.

File: extension/src/debugger/languages/dotnet.ts (hunk 8 above)
Comment (adamint) [reply extension/src/debugger/languages/dotnet.ts:620]: Fixed in 6871335033. The ordinary-launch test now retains the logger stub and asserts the exact project/run label plus diagnostics, so removing the wiring fails the focused test.
